scripts/run-integration-tests.sh

```diff
@@ -1,18 +1,37 @@
 #!/usr/bin/env bash
 
-set -euo pipefail
+set -Euo pipefail
+
+trap 'on_error $LINENO' ERR
 
 DIR=$(cd "$(dirname "$0")"; pwd)
+source $DIR/lib/common.sh
+source $DIR/lib/aws.sh
 source $DIR/lib/cluster.sh
 
 OS=$(go env GOOS)
-ARCH=amd64
-REGION=${AWS_REGION:-us-west-2}
+ARCH=$(go env GOARCH)
+AWS_REGION=${AWS_REGION:-us-west-2}
 K8S_VERSION=${K8S_VERSION:-1.14.6}
 PROVISION=${PROVISION:-true}
 DEPROVISION=${DEPROVISION:-true}
 BUILD=${BUILD:-true}
 
+__cluster_created=0
+__cluster_deprovisioned=0
+
+on_error() {
+    # Make sure we destroy any cluster that was created if we hit run into an
```

Comment:
Nit: if we run into

Reply:
Also, don't we want to leave behind test clusters on error for investigations in some cases?  For example if you are running the tests by hand?

Reply:
I can add back in the `if [[ "$DEPROVISION" = true ]];` condition here, sure. This was added because running into  errors in running the integration tests resulted in me having to manually clean up the created test cluster (many times, and I got tired of the manual cleanup :) )

Reply:
Makes sense, I think there is still value in giving the runner the ability to choose.

Reply:
I went ahead and pushed up a change that adds back the DEPROVISION = true check.